src/harness/BenchmarkDotNet.Extensions/BenchmarkDotNet.Extensions.csproj

```diff
@@ -7,8 +7,8 @@
     <_BenchmarkDotNetSourcesN>$([MSBuild]::NormalizeDirectory('$(BenchmarkDotNetSources)'))</_BenchmarkDotNetSourcesN>
   </PropertyGroup>
   <ItemGroup Condition="'$(BenchmarkDotNetSources)' == ''">
-    <PackageReference Include="BenchmarkDotNet" Version="0.13.1.1786" />
```

Comment:
@LoopedBard3 this are the only changes we need, please revert the changes to other files.

Full historical background:
- https://github.com/dotnet/runtime/pull/70546 introduced a change to NativeAOT which removed metadata for fields that were not reflected using annotations. This broke BDN which was missing the annotations. https://github.com/dotnet/BenchmarkDotNet/pull/2020 solved the problem by enforcing the compiler to keep all metadata for BDN, but at a cost of doing more work. This resulted in OOM (`OutOfMemoryException`) for our CI as the build agents have simply too little memory. https://github.com/dotnet/BenchmarkDotNet/pull/2046 introduced a proper fix and made this particular problem go away by applying the right annotations.
- https://github.com/dotnet/roslyn/pull/62392 introduced a regression in Roslyn, which started using `GC.AllocateUninitializedArray` for the new array operator (`new byte[]`). This API is not supported by NativeAOT and started causing new failures that we have observed after you updated the version. This was reported in https://github.com/dotnet/roslyn/issues/62864 and reverted in https://github.com/dotnet/roslyn/pull/62868. In theory all we need to do now is to bump BDN version and wait until Roslyn changes flow into SDK.

Reply:
I reverted the other files, hope that's OK @LoopedBard3 .